homeassistant/components/group/climate.py

```diff
@@ -0,0 +1,396 @@
+
+        # Temperature settings
+        self._attr_target_temperature = reduce_attribute(
+            states, ATTR_TEMPERATURE, reduce=lambda *data: mean(data)
+        )
+
+        self._attr_target_temperature_step = reduce_attribute(
+            states, ATTR_TARGET_TEMP_STEP, reduce=max
+        )
+
+        self._attr_target_temperature_low = reduce_attribute(
+            states, ATTR_TARGET_TEMP_LOW, reduce=lambda *data: mean(data)
+        )
+        self._attr_target_temperature_high = reduce_attribute(
+            states, ATTR_TARGET_TEMP_HIGH, reduce=lambda *data: mean(data)
+        )
+
+        self._attr_current_temperature = reduce_attribute(
+            states, ATTR_CURRENT_TEMPERATURE, reduce=lambda *data: mean(data)
+        )
+
+        self._attr_min_temp = reduce_attribute(states, ATTR_MIN_TEMP, reduce=max)
+        self._attr_max_temp = reduce_attribute(states, ATTR_MAX_TEMP, reduce=min)
+        # End temperature settings
+
+        # Build util that will help with the computation of union of modes
+        # across all the entity states.
+        def merge_modes(modes: list[list[Any]]) -> list[Any]:
+            return list(set().union(*modes))
+
+        # available HVAC modes
+        all_hvac_modes = list(find_state_attributes(states, ATTR_HVAC_MODES))
+        if all_hvac_modes:
+            self._attr_hvac_modes = merge_modes(all_hvac_modes)
+
+        current_hvac_modes = [
+            x.state
+            for x in states
+            if x.state not in [HVACMode.OFF, STATE_UNAVAILABLE, STATE_UNKNOWN]
+        ]
+        # return the most common hvac mode (what the thermostat is set to do) except OFF, UNKNOWN and UNAVAILABE
+        if current_hvac_modes:
+            self._attr_hvac_mode = HVACMode(
+                max(sorted(set(current_hvac_modes)), key=current_hvac_modes.count)
+            )
+        # return off if any is off
+        elif any(x.state == HVACMode.OFF for x in states):
+            self._attr_hvac_mode = HVACMode.OFF
+        # else it's none
+        else:
+            self._attr_hvac_mode = None
+        # return the most common action if it is not off
+        hvac_actions = list(find_state_attributes(states, ATTR_HVAC_ACTION))
+        current_hvac_actions = [a for a in hvac_actions if a != HVACAction.OFF]
+        # return the most common action if it is not off
+        if current_hvac_actions:
+            self._attr_hvac_action = max(
+                sorted(set(current_hvac_actions)), key=current_hvac_actions.count
+            )
+        # return action off if all are off
+        elif all(a == HVACAction.OFF for a in hvac_actions):
+            self._attr_hvac_action = HVACAction.OFF
+        # else it's none
+        else:
+            self._attr_hvac_action = None
+
+        # available swing modes
+        all_swing_modes = list(find_state_attributes(states, ATTR_SWING_MODES))
+        if all_swing_modes:
+            self._attr_swing_modes = merge_modes(all_swing_modes)
```

Comment:
```suggestion
        self._attr_swing_modes = list(set(find_state_attributes(states, ATTR_SWING_MODES)))
```
I suppose something like this should do? (and the `merge_modes` function is not needed?)

Reply:
Note that `find_state_attributes` would return a list (one entry for each state) of lists (with all the modes). The suggested changed would yield a list of lists (note that lists are hashable and can be used inside a set). For that reason the logic `list(set().union(*modes))` is used. Wrapped it in a simple method for readability purposes given that it is also being used multiple times.

Reply:
Right, my point was more that this needs to be able to be set as an empty list.
Now if nothing comes back it will just skip it and `swing_modes` will be the previous list.

Reply:
Done